### DISCUSSION TITLE
pytest/sbase running tests locally do not load image in chrome, if open URL manually in browser, the images are loaded/displayed

### DISCUSSION THREAD

**Discussion by ayimaok** [seleniumbase/SeleniumBase#2436] [Q&A] [answered]
I am running tests locally and in Gitlab CI, in firefox everything is good, in Chrome, the 3D image is not displayed, if I manually open Chrome browser on my machine, using the same URL, the 3D images are loaded and displayed.

**Comment by mdmintz** [comment, marked as answer]
Try “headless2” mode or “xvfb” mode. The default setting on Linux is regular “headless” mode, which might be slightly different, although faster than headed mode.

**Reply by ayimaok** [reply]
I would like to run these tests in headed mode locally, but also headless in the CI. 
Both are giving the same error, either in Chrome headed or headless. "pytest tests/testFile.py --chrome" locally or "pytest path/to/test/ --headless" 
If running tests locally, should "headless2" be run in the command line? Sorry a beginner.

**Reply by ayimaok** [reply]
Sorry, tried running "headless2" and "xvfb", did not help. It is fine using --firefox or --safari, but not in chrome, the 3D image and its container are not displayed

**Reply by mdmintz** [reply]
Would be --headless2 or --xvfb from pytest for Linux.

**Comment by ayimaok** [comment]
@mdmintz, no, when running tests using seleniumbase in Chrome the 3D graphics are absent, not displayed. When running tests using Selenium without seleniumbase, the 3D image is displayed. Neither --headless2 nor --xvfb are working. Would you have any other suggestion? Could it be possible that some version of Google Chrome is not compatible with the current version of seleniumbase?

**Reply by mdmintz** [reply]
The 3D graphics are disabled by default for performance. You can enable them with `--enable-3d-apis`:

https://github.com/seleniumbase/SeleniumBase/blob/f44d80369e9da98b288e2982a3a6d6f2c369a5e4/README.md?plain=1#L707

**Comment by ayimaok** [comment]
@mdmintz, I tried this too, does not work. 
`pytest -s -v check_pages/page_dom_check.py  --number 20 --params resources/SSCX_Portal/sscx_dom_check.json --domain https://sscx-portal.kcp.bbp.epfl.ch/sscx-portal  --enable-3d-apis --browser firefox `

On MacBookAir 14.2, the above works on firefox, but not on chrome, as said, in headed or headless, --headless2 state with or without --enable-3d-apis.

**Reply by mdmintz** [reply]
There's also `--swiftshader`:

https://github.com/seleniumbase/SeleniumBase/blob/f44d80369e9da98b288e2982a3a6d6f2c369a5e4/README.md?plain=1#L708

**Comment by ayimaok** [comment]
Does not work. I changed to running in Gitlab using firefox/geckodriver and now get HTTP method not allowed.

**Reply by mdmintz** [reply]
If you want to fully customize your driver options, use [Syntax Format 9](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#-9-overriding-the-driver-via-basecase) with the settings you need.
